datameta/api/users.py

```diff
@@ -88,34 +89,26 @@ def put(request: Request):
     if target_user is None:
         raise HTTPNotFound() # 404 User ID not found
 
-    has_group_rights = auth_user.group_admin and auth_user.group.uuid == target_user.group.uuid
-    has_admin_rights = auth_user.site_admin or has_group_rights
-    edit_own_user = auth_user.uuid == target_user.uuid
-
     # First, check, if the user has the rights to perform all the changes they want
 
     # The user has to be site admin to change another users group
-    if group_id is not None and not auth_user.site_admin:
```

Comment:
Why do you remove "is not None" in this case but not in the following cases?

Reply:
Because I am a consistency wizard XD

But seriously, for the subsequent cases, you cannot ask `if <variable>`, as they could be set to `False` (for disabling user/revoking status). In case of the group_id this also catches the hypothetical scenario of an empty group_id (by silently suppressing it, which is maybe also not the best solution).